addon/components/polaris-popover.js

```diff
@@ -90,11 +93,73 @@ export default Component.extend({
   /*
    * Internal properties.
    */
+  verticalPosition: computed('preferredPosition', function() {
+    let preferredPosition = this.get('preferredPosition');
+
+    if (preferredPosition === 'above') {
+      return preferredPosition;
+    }
+
+    if (preferredPosition === 'below') {
+      return preferredPosition;
+    }
+
+    // If set to `mostSpace`, the value will be calculated
+    // and set when the user opens the popover.
+
+    return null;
+  }),
+
   triggerStyle: computed(function() {
     return htmlSafe(`
       display: inline-block;
       overflow: inherit;
       border: none;
     `);
   }).readOnly(),
+
+  /**
+   * Checks the dropdown trigger's location on
+   * screen to determine which vertical direction
+   * has more space to open the dropdown.
+   */
+  mostVerticalSpace() {
+    let component;
+
+    // Hack to get the component's container
+    // element since this component uses `tagName: ''`
+    if (ViewUtils && ViewUtils.getViewBounds) {
+      component =  ViewUtils.getViewBounds(this).parentElement;
+    } else {
+      component = this._renderNode.contextualElement;
+    }
+
+    let triggers = component.querySelectorAll('.ember-basic-dropdown-trigger');
+
+    if (triggers.length > 1) {
+      warn('Multiple popover triggers found. Defaulting to `preferredPosition` of `below`');
+      return 'below';
+    }
+
+    let [trigger] = triggers;
+    let { top, bottom } = trigger.getBoundingClientRect();
+    let windowHeight = window.innerHeight;
+    let bottomSpace = windowHeight - bottom;
+
+    // Use `below` if distance is equal
+    return bottomSpace >= top ? 'below' : 'above';
+  },
+
+  actions: {
+    onOpen() {
+      // Check to see if `preferredPosition` is set to `mostSpace`
+      // onOpen, since user could have scrolled vertically since
+      // the time the component originally rendered.
+      let preferredPosition = this.get('preferredPosition');
+
+      if (preferredPosition === 'mostSpace') {
+        this.set('verticalPosition', this.mostVerticalSpace());
```

Comment:
Not a huge fan of the naming of `mostVerticalSpace` as a method - it sounds more like a property since there's no verb in the name. How about `getMostVerticalSpace` or making it a `volatile` CP and keeping the `mostVerticalSpace` name?